src/lib/context/shared/context.service.ts

```diff
@@ -58,6 +74,29 @@ export class ContextService {
     this.context$.next(context);
   }
 
+  private updateContextWithQueryParams(context: DetailedContext) {
+    return this.route.queryParams
```

Comment:
If I put this in constructor, it will be overwritten after the context is loaded